### PR TITLE
Add an application_name session setting

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Depends on https://github.com/crate/crate/pull/12835

The PostgreSQL JDBC client sets this implicitly. So far this resulted in
a warning in the logs

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -59,6 +59,9 @@ Deprecations
 Changes
 =======
 
+- Added a ``application_name`` session setting that can be used to identify
+  clients or applications which connected to a CrateDB node.
```
Comment (matriv) [inline docs/appendices/release-notes/unreleased.rst:63]: maybe 
```suggestion
  clients or applications which connect to a CrateDB node.
```

File: docs/config/session.rst [hunk 2]
```diff
@@ -63,6 +63,17 @@ Supported session settings
...
      it is already explicitly in the schema configuration.
 
+.. _conf-session-application-name:
+
+**application_name**
+  | *Default:* ``null``
+  | *Modifiable:* ``yes``
+
+  An arbitrary application name that can be set to identify an application that
+  connected to a CrateDB node.
```
Comment (matriv) [inline docs/config/session.rst:73]: Also here, but maybe you used past tense for a reason?
```suggestion
  connects to a CrateDB node.
```

**Comment by proddata** [comment]
Would be awesome if this could be displayed in the sys.jobs_logs in the future (or maybe some more / all session settings) 🙈

**Review by matriv — APPROVED** [review]
LGTM

**Comment by mfussenegger** [comment]
> Would be awesome if this could be displayed in the sys.jobs_logs in the future (or maybe some more / all session settings) see_no_evil

yes I had something like that in mind.

**Pull request merged**